actors/builtin/miner/miner_state.go

```diff
@@ -965,25 +989,24 @@ func (st *State) AddLockedFunds(store adt.Store, currEpoch abi.ChainEpoch, vesti
 		vestThisTime := big.Sub(targetVest, vestedSoFar)
 		vestedSoFar = targetVest
 
-		// Load existing entry, else set a new one
-		key := EpochKey(vestEpoch)
-		lockedFundEntry := big.Zero()
-		_, err = vestingFunds.Get(key, &lockedFundEntry)
-		if err != nil {
-			return err
-		}
-
-		lockedFundEntry = big.Add(lockedFundEntry, vestThisTime)
-		err = vestingFunds.Set(key, &lockedFundEntry)
-		if err != nil {
-			return err
+		// epoch already exists. Load existing entry
+		// and update amount.
+		if index, ok := epochToIndex[vestEpoch]; ok {
+			currentAmt := vestingFunds.Funds[index].Amount
+			vestingFunds.Funds[index].Amount = big.Add(currentAmt, vestThisTime)
+		} else {
+			// create a new entry and insert at the correct position
+			// st.VestingFunds is sorted by epoch
+			entry := VestingFund{Epoch: vestEpoch, Amount: vestThisTime}
+			vestingFunds.Funds = append(vestingFunds.Funds, entry)
```

Comment:
By not updating the `epochToIndex` map, this implicitly assumes that all the `vestEpoch` will be unique. Given the quantization, I don't think we can be sure of that, so update the mapping in case the same epoch appears again. Please test this with a quant spec that has a step much smaller than the quantization.

Reply:
Done.